src/OpenTelemetry.Api/Internal/Guard.cs

```diff
@@ -19,7 +19,7 @@
 using System.Runtime.CompilerServices;
 using System.Threading;
 
-#if !NETCOREAPP3_0_OR_GREATER
+#if !NET6_0_OR_GREATER
```

Comment:
Nothing that needs to be addressed in this PR, but this directive gives me some pause. Here's the targets the API project are compiled for

https://github.com/open-telemetry/opentelemetry-dotnet/blob/51c6d13460b6ccca49d1792d82bf656b8bc9d48d/src/OpenTelemetry.Api/OpenTelemetry.Api.csproj#L4

This directive is always true as compiled by the API project. I understand this file is referenced from other projects where the directive may be false, but I'm kind of surprised this hasn't caused us some issues.

I guess it's not a problem because our implementation is internal and the [real implementation](https://docs.microsoft.com/dotnet/api/system.runtime.compilerservices.callerargumentexpressionattribute) is not?

Even if it's not necessarily causing us problems, I think it may make sense to add a net6.0 target to the API and SDK projects so that net6.0+ applications get a reference to the API/SDK packages that do not contain unnecessary code.

Reply:
Good point - will revisit this as a follow up.